### PR TITLE
Always ask controller about models to migrate.

### PR DESCRIPTION
**Pull request by howbazaar**
Rather than rely on the admin's locally cached info, always ask the controller about the models when determining the model uuid to migrate.

### CODE REVIEW

**Comment by howbazaar** [comment]
## QA

Bootstrapped two lxd controllers, source and target
Added a user to both of them
For the user, gave it a different XDG_DATA_HOME, and registered the users and added a model.
Then from the admin's source, set up the migration.
It migrated.

File: cmd/juju/commands/migrate.go [hunk 1]
```diff
@@ -147,6 +148,45 @@ func (c *migrateCommand) Run(ctx *cmd.Context) error {
...
+	models, err := api.AllModels()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	// Look for the uuid based on name. If the model name doesn't container a
+	// slash, then only accept the model name if there exists only one model
+	// with that name.
+	owner := ""
+	name := c.model
+	if strings.Contains(name, "/") {
```
Comment (mjs) [inline cmd/juju/commands/migrate.go]: There's no helper for this already?

File: cmd/juju/commands/migrate.go [hunk 2]
```diff
@@ -147,6 +148,45 @@ func (c *migrateCommand) Run(ctx *cmd.Context) error {
...
+	}
+	var matches []base.UserModel
+	for _, model := range models {
+		if model.Name != name {
+			continue
+		}
+		if owner != "" && model.Owner != owner {
+			continue
+		}
+		matches = append(matches, model)
```
Comment (mjs) [inline cmd/juju/commands/migrate.go]: I would have done:

```go
if model.Name == name && (owner == "" || model.Owner == owner) {
     matches = append(matches, model)
}
```

perhaps with a helper for the owner matching part.

File: cmd/juju/commands/migrate.go (hunk 1 above)
Comment (howbazaar) [inline cmd/juju/commands/migrate.go]: It appears not.

**Comment by howbazaar** [comment]
$$merge$$

**Comment by howbazaar** [comment]
intermittent failure: 
FAIL: pinger_test.go:104: pingerSuite.TestAgentConnectionDelaysShutdownWithPing

$$merge$$

**Pull request merged**